fhir-search/src/main/java/com/ibm/fhir/search/parameters/ParametersMap.java

```diff
@@ -97,12 +71,12 @@ public void insert(String code, SearchParameter parameter) {
      * @implSpec package-private to prevent insertion from outside the package
      */
     public void insertAll(ParametersMap map) {
-        for (Entry<String, SearchParameter> entry : map.codeMap.entrySet()) {
-            insert(entry.getKey(), entry.getValue());
+        for (SearchParameter sp : map.urlMap.values()) {
+            insert(sp.getCode().getValue(), sp);
         }
     }
 
-    public SearchParameter lookupByCode(String searchParameterCode) {
```

Comment:
Minor. This Set is mutable. Could make this immutable on return. That may have performance implications given how much this stuff gets used. Perhaps use a custom class which is a HashSet and implements an interface with only the getters you need.

Reply:
Good call.  I think wrapping the set via Collections.immutableSet should be fine.

Reply:
done